src/imap.rs

```diff
@@ -596,7 +624,7 @@ impl Imap {
     }
 
     pub fn fetch(&self, context: &Context) -> libc::c_int {
-        if !self.is_connected() {
+        if !self.is_connected() || !context.sql.is_open() {
```

Comment:
don't do anything if we have no sql -- maybe it'd be better to have a ``context.is_terminating()`` bool.